massgov/mayflower
# DP-16164 Adjustable key message container height

Any PRs being created needs a changelog.txt file before being merged into dev. See: [Change Log Instructions](https://github.com/massgov/mayflower/blob/dev/docs/change-log-instructions.md)


## Description

- Remove min-height from the elements in the header key message overlay container
- Make the container height adjustable based on available content and add no extra spacing below the title.
      - Set `.ma__key-message__title`'s `margin-bottom` 0 to avoid adding extra spacing when no sub title, description or button to display.
      - Add 20 px `margin-top` to `.ma__key-message__content` to substitute the removed `.ma__key-message__title`'s `margin-bottom`.
      - Set `margin-top` 40px to `.ma__key-message__cta` when it follows `.ma__key-message__title` immediately (= no sub title or description).


## Related Issue / Ticket

- [JIRA issue](https://jira.mass.gov/browse/DP-16164)


## Steps to Test


1. Go to http://localhost:3000/patterns/05-pages-marketing-campaign/05-pages-marketing-campaign.html

2. In patternlab/styleguide/source/_patterns/05-pages/marketing-campaign.json, empty the following values and see the container maintains even padding for top and bottom:
     - subtitle (preContent > keyMessage > subtitle)
     - description (preContent > keyMessage > description) * leave `{}`.
     - button text (preContent > keyMessage > button > text)

**Background + title**
[desktop] 
<img width="801" alt="bg_title" src="https://user-images.githubusercontent.com/9633303/66948212-40e7d400-f022-11e9-8334-ec6dbf07d67e.png">
[mobile]
<img width="245" alt="bg_title-mobile" src="https://user-images.githubusercontent.com/9633303/66948222-46451e80-f022-11e9-8710-31d2af0f9652.png">
[tablet]
<img width="462" alt="bg-title-tablet" src="https://user-images.githubusercontent.com/9633303/66948246-51984a00-f022-11e9-9862-57fd0b11cd70.png">

**Background + title + button**
[desktop] 
<img width="798" alt="bg_title_button" src="https://user-images.githubusercontent.com/9633303/66948767-4f82bb00-f023-11e9-9105-96bb9418d6ef.png">
[mobile]
<img width="326" alt="bg_title_button-mobile" src="https://user-images.githubusercontent.com/9633303/66948739-4396f900-f023-11e9-94d0-60cce4f67e42.png">
[tablet]
<img width="424" alt="bg_title_button-tablet" src="https://user-images.githubusercontent.com/9633303/66948569-f6b32280-f022-11e9-9a74-375526af40f1.png">

3. Go to http://localhost:3000/patterns/05-pages-marketing-campaign-as-solid/05-pages-marketing-campaign-as-solid.html.

4. In patternlab/styleguide/source/_patterns/05-pages/marketing-campaign~as-solid.json, empty the following values and see the container maintains even padding for top and bottom:
     - subtitle (preContent > keyMessage > subtitle)
     - description (preContent > keyMessage > description) * leave `{}`.
     - button text (preContent > keyMessage > button > text)

**Solid bg color + title**
[desktop] 
<img width="802" alt="solid_title" src="https://user-images.githubusercontent.com/9633303/66949242-2a427c80-f024-11e9-825c-b1913a3a3a3a.png">
[mobile] 
<img width="280" alt="solid_title-mobile" src="https://user-images.githubusercontent.com/9633303/66949287-3fb7a680-f024-11e9-9bdc-709b85da5a6f.png">
[tablet]
<img width="515" alt="solid_title-tablet" src="https://user-images.githubusercontent.com/9633303/66949300-434b2d80-f024-11e9-9085-19be49df6910.png">

**Solid bg color + title + button**
[desktop] 
<img width="802" alt="solid_title_button" src="https://user-images.githubusercontent.com/9633303/66949359-5d850b80-f024-11e9-9614-15ac5dcc0eb2.png">
[mobile] 
<img width="300" alt="solid_title_button-mobile" src="https://user-images.githubusercontent.com/9633303/66949348-5a8a1b00-f024-11e9-8546-d3e7c775f903.png">

[tablet]
<img width="431" alt="soild_title_button_tablet" src="https://user-images.githubusercontent.com/9633303/66949342-56f69400-f024-11e9-856b-e3486a7dedd0.png">
 

## Screenshots
Use something like [licecap](http://www.cockos.com/licecap/) to capture gifs to demonstrate behaviors.


## Additional Notes:

Anything else to add?

#### Impacted Areas in Application


* 

#### @TODO


*

#### Today I learned...
